cloud-checksum/src/checksum/aws_etag.rs

```diff
@@ -5,12 +5,73 @@
 use crate::checksum::standard::StandardCtx;
 use crate::error::Error::ParseError;
 use crate::error::{Error, Result};
+use crate::io::Provider;
+use phf::phf_ordered_map;
 use std::cmp::Ordering;
 use std::fmt::{Display, Formatter};
 use std::hash::{Hash, Hasher};
 use std::str::FromStr;
 use std::sync::Arc;
 
+/// Defines the "best" order for part sizes that should be preferenced when copying/generating
+/// new checksums. This list takes into account defaults that are likely to show up in the AWS CLI
+/// and SDKs.
+pub static PART_SIZE_ORDERING: phf::OrderedMap<u64, u64> = phf_ordered_map! {
```

Comment:
This feels like an odd way of specifying an ordered list? An array?

Reply:
Ah.. I see - the main thing is to lookup part sizes and rank them.. all good

Reply:
Haha yes this is probably a bit overkill... It's a compile-time map that allows finding the the position of a part size using indexing. An array with a `.find(|elem| elem == target_part_size)` would be totally okay as well.